components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_icon.jsx

```diff
@@ -25,6 +28,28 @@ export default class SidebarChannelButtonOrLinkIcon extends React.PureComponent
         teammateIsBot: PropTypes.bool,
     };
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            svgError: false,
+            botIconUrl: null,
+        };
+    }
+
+    onSvgLoadError = () => {
+        this.setState({
+            svgError: true,
```

Comment:
@renilJoseph rather than keeping track of `svgError` as `true` or `false` we can keep track of the failed `url` of the boticon, as all this is tracking is the url that failed when we attempted to fetch it. And you can remove `this.state. botIconUrl` altogether.

Just make sure it works when you start with no bot icon, set the bot icon (make sure webapp updates), remove the bot icon (make sure the web app reverts to default bot icon)

Reply:
tested all 3 conditions. Works good.